lib/evmone/baseline_instruction_table.cpp

```diff
@@ -7,21 +7,43 @@
 
 namespace evmone::baseline
 {
-const CostTable& get_baseline_cost_table(evmc_revision rev) noexcept
+namespace
 {
-    static constexpr auto cost_tables = []() noexcept {
-        std::array<CostTable, EVMC_MAX_REVISION + 1> tables{};
-        for (size_t r = EVMC_FRONTIER; r <= EVMC_MAX_REVISION; ++r)
+constexpr auto common_cost_tables = []() noexcept {
+    std::array<CostTable, EVMC_MAX_REVISION + 1> tables{};
+    for (size_t r = EVMC_FRONTIER; r <= EVMC_MAX_REVISION; ++r)
+    {
+        auto& table = tables[r];
+        for (size_t i = 0; i < table.size(); ++i)
         {
-            auto& table = tables[r];
-            for (size_t i = 0; i < table.size(); ++i)
-            {
-                table[i] = instr::gas_costs[r][i];  // Include instr::undefined in the table.
-            }
+            table[i] = instr::gas_costs[r][i];  // Include instr::undefined in the table.
         }
-        return tables;
-    }();
+    }
+    return tables;
+}();
 
-    return cost_tables[rev];
+constexpr auto legacy_cost_tables = []() noexcept {
+    auto tables = common_cost_tables;
+    tables[EVMC_CANCUN][OP_RJUMP] = instr::undefined;
+    tables[EVMC_CANCUN][OP_RJUMPI] = instr::undefined;
```

Comment:
This has to also undefine RJUMPV, CALLF, RETF

Reply:
Ok. Will modify corresponding commits